### ISSUE TITLE
ReferenceError: Can't find variable: Promise

### ISSUE THREAD

**Issue by ascot21** [cibernox/ember-power-select#933]
I get the error `ReferenceError: Can't find variable: Promise` when I use `selectChoose` in an acceptance test.  It runs fine in the browser but fails in the console.

I'm using `Ember 2.11.1`, `Ember Power Select 1.8.5` and `Phantom JS 2.1.1`.

**Comment by danielleadams** [comment]
@ascot21 I ran into this today, and I found that a (temporary) workaround is to downgrade to `1.6.1`.

**Comment by luketheobscure** [comment]
I haven't run across this problem, but have you  tried including the babel polyfill? Just add this to your `EmberApp` arguments in `ember-cli-build.js`:

```
'ember-cli-babel': {
  includePolyfill: true
}
```

**Comment by danielleadams** [comment]
That'll do it! My initial instinct was that it was babel-related, but I must not have configured the polyfill correctly in the build file. Thanks @luketheobscure!

**Comment by cibernox** [comment]
I've been on holidays all last week, Im still catching up. 

On on hand, phantomjs is officially dead. I'd encourage people to start testing their apps in browsers people really use, like chrome headless or firefox.
On the other hand, I don't expect people to include the polyfill in order to make this addon work. I'll dig into it. I should use `RSVP.Promise` everywhere, not the native one.

**Comment by cibernox** [comment]
Ok, I've found the cause. It turns out it is a problem with promises and regenerator.

```js
  function _asyncToGenerator(fn) {
    return function () {
      var gen = fn.apply(this, arguments);
      return new Promise(function (resolve, reject) { // <===== here!
        function step(key, arg) {
          try {
            var info = gen[key](arg);
            var value = info.value;
          } catch (error) {
            reject(error);
            return;
          }
         // ... more code
      });
    };
  }
```

In recent versions of EPS I started using async await on tests to people can do things like `await selectChoose('#my-select', 'Barry')`. But while I take care of conditionally import the `regenerator` runtime in browsers that don't support generators, seems that the runtime relies on the existence of a global promise.

I'll investigate if there is a workaround. In the meanwhile my advice is:

- Kill PhantomJS with 🔥 . It is a terrible, bug-ridden browser that literally 0 ppl in the world use.
- Include the babel polyfill, perhaps conditionally only in testing. This problem doesn't affect the behaviour of the select in browsers without promises like IE, only affects testing.
